xformers/benchmarks/benchmark_mem_eff_attention.py

```diff
@@ -49,38 +52,60 @@ def ref_attention(q, k, v, attn_bias=None, p=0.0):
     return attn @ v
 
 
+def ref_attention(q, k, v, attn_bias, p=0.0):
+    assert q.ndim == 4
+
+    def T(t):
+        return t.permute((0, 2, 1, 3)).reshape(
+            [t.shape[0] * t.shape[2], t.shape[1], t.shape[3]]
+        )
```

Comment:
nit: I wonder if we should still keep the previous benchmark without the permute + contiguous in the hot path.

The improved benchmark you just added is very important as it's the main entry point for the users, but it might also hide some potential improvements to be done in the standard attention because now we are also measuring those extra overheads.

I would maybe this benchmark on top of the previous one, so that we have more numbers when measuring things together. But this is only a suggestion